packages/firebase_database/lib/ui/client_sorted_list.dart

```diff
@@ -17,26 +14,20 @@ import 'utils/stream_subscriber_mixin.dart';
 // We can be smarter about how we handle insertion and keep the list always
 // sorted. See example here:
 // https://github.com/firebase/FirebaseUI-iOS/blob/master/FirebaseDatabaseUI/FUISortedArray.m
-class FirebaseSortedList extends ListBase<DataSnapshot> with StreamSubscriberMixin<Event> {
-  FirebaseSortedList({
-    @required this.query,
+class ClientSortedList extends FirebaseList {
+  ClientSortedList({
+    @required Query query,
     @required this.comparator,
     this.onChildAdded,
     this.onChildRemoved,
     this.onChildChanged,
-    this.onValue,
-  }) {
-    assert(query != null);
+  }) : super(query: query) {
     assert(comparator != null);
     listen(query.onChildAdded, _onChildAdded);
     listen(query.onChildRemoved, _onChildRemoved);
     listen(query.onChildChanged, _onChildChanged);
```

Comment:
Could you cancel the subscriptions on disposal? setState is going to be called after this widget has been disposed.